modules/generator/processor/spanmetrics/spanmetrics.go

```diff
@@ -0,0 +1,128 @@
+package spanmetrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+
+	gen "github.com/grafana/tempo/modules/generator/processor"
+	processor_util "github.com/grafana/tempo/modules/generator/processor/util"
+	"github.com/grafana/tempo/pkg/tempopb"
+	v1_trace "github.com/grafana/tempo/pkg/tempopb/trace/v1"
+	tempo_util "github.com/grafana/tempo/pkg/util"
+)
+
+type processor struct {
+	cfg Config
+
+	spanMetricsCallsTotal      *prometheus.CounterVec
+	spanMetricsDurationSeconds *prometheus.HistogramVec
+
+	// for testing
+	now func() time.Time
+}
+
+func New(cfg Config, tenant string) gen.Processor {
+	return &processor{
+		cfg: cfg,
+		now: time.Now,
+	}
+}
+
+func (p *processor) Name() string { return Name }
+
+func (p *processor) RegisterMetrics(reg prometheus.Registerer) error {
+	labelNames := []string{"service", "span_name", "span_kind", "span_status"}
+	if len(p.cfg.Dimensions) > 0 {
+		labelNames = append(labelNames, p.cfg.Dimensions...)
+	}
+
+	p.spanMetricsCallsTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "traces",
```

Comment:
We are using the `traces` namespace here to match with the metrics generated by the Grafana Agent. If we generate the same metrics, it will be easier for people to reuse queries/dashboards for both.
The service graphs view in Grafana for instance expects metrics with the name `traces_service_graph_...`. We could make this configurable in Grafana, but for now we want to keep it as simple as possible.

Reply:
Hmm. Yeah I wish we could make this `tempo` but I don't see a clear path to making that change. I guess its ok to leave it as is.